app/graphql/types/git/commit_type.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/BlockLength
+Types::Git::CommitType = GraphQL::ObjectType.define do
+  # rubocop:enable Metrics/BlockLength
+  name 'Commit'
+  description 'A commit of a repository'
+
+  field :id, !types.ID do
+    description 'The sha hash of the commit'
+  end
+
+  field :parentIds, !types[!types.ID] do
+    description 'The parents of the commit'
+    property :parent_ids
+  end
+
+  field :message, !types.String do
+    description 'The commit message'
+  end
+
+  field :author, !Types::Git::UserType do
+    description 'The author of the commit'
+
+    resolve(lambda do |commit, _arguments, _context|
+      GitUser.new(commit.author_name, commit.author_email)
+    end)
+  end
+
+  field :authoredAt, !Types::TimeType do
+    description 'The time the commit was authored at'
+    property :authored_date
+  end
+
+  field :committer, !Types::Git::UserType do
+    description 'The committer of the commit'
+
+    resolve(lambda do |commit, _arguments, _context|
+      GitUser.new(commit.committer_name, commit.committer_email)
+    end)
+  end
+
+  field :committedAt, !Types::TimeType do
+    description 'The time the commit was created'
+    property :committed_date
+  end
+
+  field :refNames, !types[!types.String] do
+    description 'The names of the references that point to this commit'
+    property :ref_names
+  end
+
+  field :directory, !types[!Types::Git::DirectoryEntryType] do
```

Comment:
What happens if the given path is non-existant, or if it points to a file? Maybe the type `[DirectoryEntry!]` (with the option to return `null`) would be better here.